core/tests/protractor/accessibility.js

```diff
@@ -19,28 +19,158 @@
+    await browser.actions().sendKeys(protractor.Key.TAB).perform();
+
+    // Should move the focus back to the elementToFocus.
+    await browser.actions().sendKeys(key).perform();
+    expect(await elementToFocus.getAttribute('id')).toEqual(
+      await (await browser.driver.switchTo().activeElement())
+        .getAttribute('id'));
+  };
+
   it('should skip to the main content element', async function() {
     await libraryPage.get();
     await browser.actions().sendKeys(protractor.Key.TAB).perform();
-    var skipLink = element(by.css('.protractor-test-skip-link'));
     await waitFor.elementToBeClickable(skipLink, 'Could not click skip link');
     await skipLink.click();
-    var mainContent = element(by.css('.protractor-test-main-content'));
     expect(await mainContent.getAttribute('id')).toEqual(
       await (await browser.driver.switchTo().activeElement())
         .getAttribute('id'));
   });
 
+  it('should move focus to the skip button in the library page',
+    async function() {
+      // Should test the skip to main content shortcut.
+      await libraryPage.get();
+      await checkActionShortcuts('s', skipLink);
+    });
+
+  it('should move focus to the search bar in the library page',
+    async function() {
+      // Should move the focus to the search bar.
+      await libraryPage.get();
+      await checkActionShortcuts('/', searchBar);
+    });
+
+  it('should move focus to the category bar in library page',
+    async function() {
+      // Should move the focus to the category bar.
+      await libraryPage.get();
+      await checkActionShortcuts('c', categoryBar);
+    });
+
+  it('should move focus to skip to main content button in exploration player',
+    async function() {
+      // Should Create a user and login.
+      await users.createUser('user11@accessibility.com', 'user11accessibility');
+      await users.login('user11@accessibility.com', true);
+
+      // Should create and play a dummy exploration.
+      await workflow.createAndPublishTwoCardExploration(
+        EXPLORATION.title,
+        EXPLORATION.category,
+        EXPLORATION.objective,
+        EXPLORATION.language
+      );
+      await libraryPage.get();
+      await libraryPage.findExploration('A new exploration');
+      await libraryPage.playExploration('A new exploration');
+
+      // Should test the skip to main content shortcut.
+      await checkActionShortcuts('s', skipLink);
+    });
+
+  it('should move focus to next and back buttons in exploration player',
+    async function() {
+      // Play a dummy exploration.
+      await libraryPage.get();
+      await libraryPage.findExploration('A new exploration');
+      await libraryPage.playExploration('A new exploration');
+
+      // Should test the continue button shortcut.
+      await waitFor.elementToBeClickable(continueButton);
+      await checkActionShortcuts('j', continueButton);
+      await browser.actions().sendKeys(protractor.Key.ENTER).perform();
```

Comment:
Can you check that the continue action actually gets triggered? Maybe check that the next page in the exploration appears. Ditto for the previous and next buttons below.

Reply:
Could you add a comment explaining this? We're assuming that we start at the beginning of the exploration, then click "continue", "back", and "next", in that order. It would be really easy for someone to add a step in-between those (e.g. a "next" to the start) and not realize they're breaking your assumptions (because then the card where you click "continue" would have a "back" button).

Reply:
I have added the comment